api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubCommitV3.java

```diff
@@ -0,0 +1,345 @@
+                String committerLogin = (committerUserObject == null) ? "unknown" : restClient.getString(committerUserObject, "login");
+                commit.setScmCommitterLogin(committerLogin);
+            }
+
+            int numberChanges = 0;
+            if (cObj.get("added") instanceof List) {
+                numberChanges += ((List) cObj.get("added")).size();
+            }
+            if (cObj.get("removed") instanceof List) {
+                numberChanges += ((List) cObj.get("removed")).size();
+            }
+            if (cObj.get("modified") instanceof List) {
+                numberChanges += ((List) cObj.get("modified")).size();
+            }
+
+            commit.setNumberOfChanges(numberChanges);
+            setCommitPullNumber(commit);
+            setCollectorItemId(commit);
+            commitsList.add(commit);
+        }
+
+        // For a merge commit for "Rebase and Merge"
+        setCommitPullNumbersForRebaseAndMergeCommit(commitsList);
+
+        return commitsList;
+    }
+
+    protected void setCommitPullNumbersForRebaseAndMergeCommit(List<Commit> commitsList) {
+        List<Commit> commitsWithPullNumber = Optional.ofNullable(commitsList)
+                                            .orElseGet(Collections::emptyList).stream()
+                                            .filter(commit -> !StringUtils.isEmpty(commit.getPullNumber()))
+                                            .collect(Collectors.toList());
+
+        // In case of Rebase and Merge, only the last commit in the list of commits on the "merge commit" json should have the PR number.
+        // This is because, there should be a corresponding PR in the DB whose "merge_commit_sha" matches the last commit in the list of commits
+        if (checkCommitsWithPullNumber(commitsWithPullNumber)
+                && checkCommitsListForSettingPullNumber(commitsList)) {
+            Commit commitWithPR = commitsWithPullNumber.get(0);
+            commitsList.forEach(commit -> {commit.setPullNumber(commitWithPR.getPullNumber());});
+        }
+    }
+
+    private boolean checkCommitsWithPullNumber(List<Commit> commitsWithPullNumber) {
+        if (!CollectionUtils.isEmpty(commitsWithPullNumber)
+                && (commitsWithPullNumber.size() == 1)) { return true; }
+
+        return false;
+    }
+
+    private boolean checkCommitsListForSettingPullNumber(List<Commit> commitsList) {
+        if (!CollectionUtils.isEmpty(commitsList) && (commitsList.size() > 1)) { return true; }
+
+        return false;
+    }
+
+    protected void setCommitPullNumber (Commit commit) {
+        GitRequest pr = gitRequestRepository.findByScmRevisionNumberOrScmMergeEventRevisionNumber(commit.getScmRevisionNumber());
+        if (pr == null) {
+            pr = gitRequestRepository.findByCommitScmRevisionNumber(commit.getScmRevisionNumber());
+        }
+        if (pr != null) {
```

Comment:
else?

Reply:
In this case if the PR is not found, then the pull number will remain unassigned or null
In other words, pull number is only assigned when a corresponding PR is found